docs/reference/migration/transient-settings-migration-guide.asciidoc

```diff
@@ -0,0 +1,120 @@
+[[transient-settings-migration-guide]]
+=== Transient settings migration guide
+
+////
+[source,console]
+----
+PUT _cluster/settings
+{
+  "transient": {
+    "cluster.indices.close.enable": false,
+    "indices.recovery.max_bytes_per_sec": "50mb"
+  }
+}
+----
+// TEST[warning:[transient settings removal] Updating cluster settings through transientSettings is deprecated. Use persistent settings instead.]
+////
+
+Transient cluster settings are deprecated. Previously, you could use transient
+settings to make temporary configuration changes to a cluster. However,
+transient settings don't survive full cluster restarts. A cluster restart can
+occur forcibly if enough master-eligible nodes fail. This can clear transient
+settings unexpectedly, leading to a potentially undesired cluster configuration.
+
+To avoid deprecation warnings, reset any transient settings you've configured on
+your cluster. Convert any transient setting you'd like to keep to a persistent
+setting, which persists across cluster restarts. You should also update any
+custom workflows and applications to use persistent settings instead of
+transient settings.
+
+IMPORTANT: Some Elastic products may use transient settings. These settings
+should not be changed.
```

Comment:
> IMPORTANT: Some Elastic products may use transient settings. These settings
should not be changed.

This is hard for a user to know, and hard for us to give good guidance. I wonder if we should be a bit more specific like:

IMPORTANT: Some Elastic products may use transient settings while performing specific operations. Only remove transient settings that were set by users or custom workflows.

I'm still not sure about that wording ... thoughts ?

Reply:
I've reworded this to:

> Some Elastic products may use transient settings when performing
specific operations. Only reset transient settings configured by you, your
users, or your custom workflows and applications.

However, I still think it'll be difficult for users to discern between Elastic- and user-managed settings here. This may be the best advice we can offer until our products migrate away from transient setting though.

Reply:
That sounds great! thanks